packages/link/src/components/FloatingLink/FloatingLinkUrlInput.tsx

```diff
@@ -31,7 +34,13 @@ export const useFloatingLinkUrlInput = (
 ) => {
   const onChange: React.ChangeEventHandler<HTMLInputElement> =
     React.useCallback((e) => {
-      floatingLinkActions.url(e.target.value);
+      const value = e.target.value;
+
+      if (isEncoded(value)) {
+        floatingLinkActions.url(value);
+      } else {
+        floatingLinkActions.url(encodeURI(value));
+      }
```

Comment:
Can you refactor the entire logic (conditionally calling `encodeURI`) into a function called `encodeUrlIfNeeded` and test that?